jicoco-metrics/src/main/kotlin/org/jitsi/metrics/MetricsContainer.kt

```diff
@@ -110,17 +113,23 @@ open class MetricsContainer @JvmOverloads constructor(
         /** the optional initial value of the metric */
         initialValue: Long = 0
     ): CounterMetric {
-        if (metrics.containsKey(name)) {
+        val newName = when (name.endsWith("_total")) {
+            true -> name
+            false -> "${name}_total".also {
+                logger.warn("Counter '$name' was renamed to '$it' to ensure consistent metric naming.")
+            }
+        }
+        if (metrics.containsKey(newName)) {
```

Comment:
If there is another metric with "name" the Prometheus lib will throw, right? What do you think about handling that case here, so it's clear that it's expected and the warning message is clear

Reply:
Yes, it throws an IllegalArgumentException. So were you thinking of trying to return a newly-registered metric, and catch the exception and deal with it?

Reply:
No, I was thinking of catching the case and throwing our own exception. Just a suggestion I'm not sure this is a good idea or worth doing.